fiat-roles/src/main/groovy/com/netflix/spinnaker/fiat/permissions/DefaultPermissionsResolver.java

```diff
@@ -57,27 +57,52 @@
   @Setter
   private ServiceAccountProvider serviceAccountProvider;
 
+  @Value("${auth.anonymous.enabled}")
+  @Setter
+  private boolean anonymousEnabled;
+
+  @Override
+  public Optional<UserPermission> resolveAnonymous() {
+    if (!anonymousEnabled) {
+      return Optional.empty();
+    }
+
+    val groups = new ArrayList<String>(0);
+    val accounts = accountProvider.getAccounts(groups);
```

Comment:
sorry because I don't have the whole project open at this moment - is accountProvider async loading all accounts and this call just operating off the cached view?

(wondering about the retry story if we are reaching directly out to other services here)